src/DynamoCore/PackageManager/PackageUploadBuilder.cs

```diff
@@ -1,205 +1,93 @@
-            var binPath = Path.Combine(rootPath, "bin");
-            var extraPath = Path.Combine(rootPath, "extra");
-
-            root = TryCreateDirectory(rootPath);
-            dyfDir = TryCreateDirectory(dyfPath);
-            binDir = TryCreateDirectory(binPath);
-            extraDir = TryCreateDirectory(extraPath);
-        }
+            var zipFile = Zip(dir);
 
-        private static void WritePackageHeader(PackageUploadRequestBody pkgHeader, DirectoryInfo rootDir)
-        {
-            // build the package header json, which will be stored with the pkg
-            var jsSer = new JsonSerializer();
-            var pkgHeaderStr = jsSer.Serialize(pkgHeader);
-
-            // write the pkg header to the root directory of the pkg
-            var headerPath = Path.Combine(rootDir.FullName, "pkg.json");
-            if (File.Exists(headerPath)) File.Delete(headerPath);
-            File.WriteAllText(headerPath, pkgHeaderStr);
+            return new PackageVersionUpload(NewRequestBody(package), zipFile.Name);
         }
 
-        private static bool IsXmlDocFile(string path, IEnumerable<string> files)
-        {
-            if (!path.ToLower().EndsWith(".xml")) return false;
-
-            var fn = Path.GetFileNameWithoutExtension(path);
+        #endregion
 
-            return
-                files.Where(x => x.EndsWith(".dll"))
-                    .Select(Path.GetFileNameWithoutExtension)
-                    .Contains(fn);
-        }
+        #region Private Class Methods
 
-        private static bool IsDynamoCustomizationFile(string path, IEnumerable<string> files)
+        private IFileInfo Zip(IDirectoryInfo directory)
         {
-            if (!path.ToLower().EndsWith(".xml")) return false;
+            IFileInfo info;
 
-            var name = Path.GetFileNameWithoutExtension(path);
-
-            if (!name.EndsWith("_DynamoCustomization")) return false;
-
-            name = name.Remove(name.Length - "_DynamoCustomization".Length);
+            try
+            {
+                info = fileCompressor.Zip(directory.FullName);
+            }
+            catch
+            {
+                throw new Exception(Properties.Resources.CouldNotCompressFile);
+            }
 
-            return
-                files.Where(x => x.EndsWith(".dll"))
-                    .Select(Path.GetFileNameWithoutExtension)
-                    .Contains(name);
-        }
+            // the file is stored in a tempt directory, we allow it to get cleaned up by the user later
+            if (info.Length > MaximumPackageSize) throw new Exception(Properties.Resources.PackageTooLarge);
 
-        public static string NormalizePath(string path)
-        {
-            return Path.GetFullPath(new Uri(path).LocalPath)
-                       .TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar)
-                       .ToUpperInvariant();
+            return info;
         }
```

Comment:
Rather than looking at the diff here, it might be more useful to look at the file, which is now rather small.